### PR TITLE
refactor(vector-store): enhance validation and initialization processes

### PR DESCRIPTION
**Pull request by PingXie**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/langchain-google-memorystore-redis-python/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: src/langchain_google_memorystore_redis/vector_store.py [hunk 1]
```diff
@@ -448,8 +465,9 @@ def from_texts(
         texts: List[str],
         embedding: Embeddings,
         metadatas: Optional[List[dict]] = None,
-        client=None,
-        index_name=None,
+        ids: Optional[List[str]] = None,
```
Comment (craig08) [inline src/langchain_google_memorystore_redis/vector_store.py:468]: nit: Optional[List[dict]] should be equivalent to List[dict] | None ([ref](https://stackoverflow.com/questions/69440494/python-3-10-optionaltype-or-type-none)). Either one is good but it'll be better to be consistent!

**Review by craig08 — APPROVED** [review]
LGTM

File: src/langchain_google_memorystore_redis/vector_store.py (hunk 1 above)
Comment (PingXie) [reply src/langchain_google_memorystore_redis/vector_store.py:475]: consolidated on Optional. I was trying to stay away from the "Optional" keyword for these semantically mandatory arguments but yeah agreed that consistency is the key.

Comment (PingXie) [reply src/langchain_google_memorystore_redis/vector_store.py:475]: agreed - consolidated on Optional

**Pull request merged**